pkg/registration/register/common.go

```diff
@@ -227,3 +236,64 @@ func (a *AggregatedHubDriver) Cleanup(ctx context.Context, cluster *clusterv1.Ma
 	}
 	return errors.NewAggregate(errs)
 }
+
+// Clients hold all client needed to connect to hub
+type Clients struct {
+	ClusterClient         clusterv1client.Interface
+	KubeClient            kubernetes.Interface
+	AddonClient           addonclient.Interface
+	ClusterInfomerFactory clusterv1informers.SharedInformerFactory
+	KubeInformerFactory   informers.SharedInformerFactory
+	AddonInformerFactory  addoninformers.SharedInformerFactory
+}
+
+func BuildClientsFromSecretOption(s SecretOption, bootstrapped bool) (*Clients, error) {
+	var kubeConfig *rest.Config
+	var err error
+	if bootstrapped {
```

Comment:
may rename the `bootstrapped` to `bootstrap`, the `bootstrapped` may have confusions here